### ISSUE TITLE
Unable to use SMB mounts requiring authentication in OC X

### ISSUE THREAD

**Issue by xieliwei** [owncloud/core#27777]
### Steps to reproduce
1. Use latest (29 Apr 17) openSuSE Tumbleweed
2. Add ownCloud testing repository for Tumbleweed
3. Go to Settings->Storage
4. Add a SMB mount that *does not* require authentication
5. Attempt to add a SMB mount that requires authentication
6. Try to access mounts in Files app

### Expected behaviour
1. "Green circle" indicator comes up on Storage page upon activating both mounts
2. Files app should not indicate any errors

### Actual behaviour
1. "Red square" error indicator comes up on Storage page upon activating mount requiring authentication
2. Trying to access authenticated mount in Files produces this error "There was an error with message: Empty response from the server. Do you want to review mount point config in admin settings page?"
3. A red bar appears on top of the Files app with this error "Some of the configured external mount points are not connected. Please click on the red row(s) for more information"

(I have tried smbclient and writing test applications using libsmbclient (in C) and libsmbclient-php, all of them work fine with authenticated SMB mounts)

(I have also verified that OC is using libsmbclient-php. Using the OC-packaed samba client produces a different problem, so it is obvious which SMB client is being used)

### Server configuration
**Operating system:** openSuSE Tumbleweed with 4.10.12-1.1-default kernel

**Web server:** Apache/2.4.25

**Database:** mysql Ver 14.14 Distrib 5.6.35, for Linux (x86_64) using EditLine wrapper

**PHP version:** PHP 5.6.30 with Zend Engine v2.6.0

**ownCloud version:** 10.0.0 RC4

**Updated from an older ownCloud or fresh install:** Updated from ownCloud 9

**Where did you install ownCloud from:** ownCloud openSuSE repository

**Signing status (ownCloud 9.0 and above):**

Old plugins and cruft from various upgrades over the years...
https://gist.github.com/xieliwei/e5ada7f2b49294e270d0f9ce6d004d71

**The content of config/config.php:**
```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/srv\/www\/owncloud",
        "dbtype": "mysql",
        "version": "10.0.0.10",
        "dbname": "owncloud",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "theme": "",
        "maintenance": false,
        "loglevel": 0,
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
        ],
        "secret": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "0": "\/tmp\/redis.sock",
            "timeout": 0,
            "dbindex": 0
        },
        "updatechecker": false,
        "trashbin_retention_obligation": "auto",
        "htaccess.RewriteBase": "\/",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "integrity.check.disabled": false,
        "appstore.experimental.enabled": true
    }
}

```
**List of activated apps:**
```
Enabled:
  - activity: 2.3.3
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.2.8
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.0
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - market: 0.1.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - conversations
  - documents
  - encryption
  - example-theme
  - external
  - files_antivirus
  - music
  - news
  - notes
  - shorten
  - tasks
  - user_external
```

**Are you using external storage, if yes which one:** SMB

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:** Not relevant

**Operating system:** Not relevant

#### Web server error log
No relevant logs generated, even in error_log

#### ownCloud log (data/owncloud.log)
No relevant logs generated. I turned logging level to "Everything", but nothing gets generated when I test out SMB mounting. Other logs do show up in owncloud.log.

**Comment by xieliwei** [comment]
A little (less-sloppy) searching on the issue tracker brought up the following issue, which seems to be an exact match of the problem. Investigating now.

#24314

**Comment by xieliwei** [comment]
Previously-mentioned issue not relevant.

1. It affects samba 3.6.x - NAS uses samba 4.4.7, and OC server uses samba 4.6.2
2. Tried workarounds listed in the following posts referenced by the issue, no change in behavior
    - http://community.netapp.com/t5/Network-Storage-Protocols-Discussions/samba-3-6-23-30-on-CentOS-gt-error-in-smbclient/td-p/118268
    - https://bugs.launchpad.net/ubuntu/+source/samba/+bug/1572122
3. In issues affected by the badlock fix bug, smbclient and libsmbclient are affected as well. They're working fine in my case, the issue only occurs within OC.

**Comment by PVince81** [comment]
@jvillafanez in case you've seen this before

**Comment by jvillafanez** [comment]
@xieliwei It's not clear for me if you've tested with smbclient from the same server ownCloud is, or from another server. Test from the same server.

You can also check with a VM in order to verify if there is a problem with your configuration. If the VM works, then there is likely a problem with your environment. You can download the VM from https://download.owncloud.org/community/production/vm/Ubuntu_14.04-owncloud-9.1.5-1.1-201704201507.ova.zip (OVA format). It's for 9.1.5, but there shouldn't be relevant changes in the code. If the 9.1.5 works you can check differences between the smb.conf files or library versions. Feel free to break the VM because you can restore it to a previous state without worring, trying to replicate what you have in your production environment.

In addition, we'll likely need more information from you (please do NOT post sensitive information): the smb.conf file, information about the target SMB server you're trying to connect (if it's any specific windows version, or if it's a samba server, versions, etc), whether if there is any SMB protocol version disabled in the server (SMBv1 might be disabled in the server), and any relevant configuration in the SMB server (for windows for example, whether it has file deduplication active in the target share, or has some password lockout policy).

**Comment by xieliwei** [comment]
Sorry that this reply took so long. It was a combination of two causes:

1. @jvillafanez, you're right, SMBv1 (or rather NT1) was being used because I did not specify "client max protocol" in smb.conf. Apparently something changed when samba got updated and either the server stopped serving NT1 clients or clients defaulted to NT1 when "client max protocol" is not specified.

2. The above was not the only cause. Apparently something changed either with OC's or libsmbclient's handling of non-domain/workgroup shares. Previously I left the domain field blank within OC's storage configuration for samba shares, but now I need to fill it in with anything (just not blank) or the mount will fail.


I did try both solutions independently at some point, but never together, until yesterday. Now I wonder why I didn't do that earlier. Hah

Anyway, it is unlikely to be an OC bug, or maybe just some side effect of some intended code changes, closing this issue.
